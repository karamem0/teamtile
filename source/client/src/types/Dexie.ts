//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export interface ArrayEntity<T> {
  id: string,
  expired: number,
  values: T[]
}

export interface ValueEntity<T> {
  id: string,
  expired: number,
  value: T
}
