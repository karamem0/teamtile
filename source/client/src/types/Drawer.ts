//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export type DrawerType =
  | 'channel'
  | 'member'
  | 'tag';
