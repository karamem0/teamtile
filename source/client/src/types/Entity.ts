//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

export enum MembershipType {
  private = 'private',
  standard = 'standard'
}

export enum VisibilityType {
  private = 'private',
  public = 'public'
}

export enum AccordionType {
  all = 'all',
  archived = 'archived',
  pinned = 'pinned'
}

export interface Channel {
  id?: string,
  displayName?: string,
  membershipType?: MembershipType,
  webUrl?: string
}

export interface Drive {
  id?: string,
  webUrl?: string
}

export interface Group {
  id?: string,
  email?: string,
  sensitivityLabel?: string
}

export interface Icon {
  id?: string,
  data?: string
}

export interface Member {
  id?: string,
  displayName?: string,
  email?: string,
  icon?: string,
  userId?: string
}

export interface Tab {
  id?: string,
  appId?: string,
  displayName?: string,
  webUrl?: string
}

export interface Team {
  id?: string,
  archived?: boolean,
  description?: string,
  displayName?: string,
  icon?: string,
  internalId?: string,
  visibility?: VisibilityType,
  webUrl?: string
}

export interface TeamInfo {
  id?: string,
  displayName?: string,
  tenantId?: string
}
