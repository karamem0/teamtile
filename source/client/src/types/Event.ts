//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

export type Event = React.SyntheticEvent | React.SyntheticEvent<unknown> | Record<string, never>;

export type EventHandler<T = never> = (
  event?: Event,
  data?: T
) => void;
