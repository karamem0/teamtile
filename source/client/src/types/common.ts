//
// Copyright (c) 2021 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/master/LICENSE
//

export interface KeyValue<K, V> {
  key: K,
  value: V
}
