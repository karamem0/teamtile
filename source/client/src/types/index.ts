//
// Copyright (c) 2021 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/master/LICENSE
//

export interface Team {
  id?: string;
  internalId?: string;
  name?: string;
  description?: string;
  visibility?: string;
  url?: string;
  icon?: string;
}

export interface TeamChannels {
  count?: number;
  nextLink?: string;
  values?: TeamChannel[];
}

export interface TeamChannel {
  id?: string;
  name?: string;
  url?: string;
}

export interface TeamMembers {
  count?: number;
  nextLink?: string;
  values?: TeamMember[];
}

export interface TeamMember {
  id?: string;
  name?: string;
  email?: string;
}

export interface TeamDrive {
  id?: string;
  url?: string;
}
