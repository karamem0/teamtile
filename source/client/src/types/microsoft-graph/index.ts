import * as microsoftGraph from '@microsoft/microsoft-graph-types';

export type MicrosoftGraphResponse = MicrosoftGraphValueResponse | MicrosoftGraphArrayResponse | MicrosoftGraphErrorResponse | string;

export interface MicrosoftGraphValueResponse extends microsoftGraph.Entity
{
  '@odata.context': string;
}

export interface MicrosoftGraphArrayResponse {
  '@odata.context': string;
  '@odata.count': number;
  '@odata.nextLink': string;
  value: microsoftGraph.Entity[];
}

export interface MicrosoftGraphBatchResponse {
  id: string;
  status: number;
  body: MicrosoftGraphResponse;
  headers: {
    [key: string]: string;
  };
}

export interface MicrosoftGraphErrorResponse {
  error: microsoftGraph.GenericError;
}
