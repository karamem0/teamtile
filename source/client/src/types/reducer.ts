//
// Copyright (c) 2021 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/master/LICENSE
//

// Types
import {
  Channel,
  Drive,
  Icon,
  Member,
  Team
} from './entity';

export interface KeyValue<K, V> {
  key: K,
  value: V
}

export type ItemKey = string;

export type ItemValue = (
  Team &
  Icon &
  {
    channels: Channel[] | undefined,
    members: (Member & Icon)[] | undefined,
    drive: Drive | undefined
  }
);

export type Item = (
  KeyValue<ItemKey, ItemValue | null> &
  {
    visible: boolean | undefined
  }
);

export interface State {
  loading: boolean,
  items: Item[]
}

export interface Action {
  type: ActionType,
  payload: unknown
}

export enum ActionType {
  PutLoading = 'PutLoading',
  PutFilter = 'PutFilter',
  PutKeys = 'PutKeys',
  PutTeams = 'PutTeams',
  PutTeamIcons = 'PutTeamIcons',
  PutChannels = 'PutChannels',
  PutMembers = 'PutMembers',
  PutMemberIcons = 'PutMemberIcons',
  PutDrives = 'PutDrives',
}
