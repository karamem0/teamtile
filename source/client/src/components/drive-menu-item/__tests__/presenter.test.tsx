//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import { Drive } from '../../../types/entity';
import DrivelMenuItem from '../presenter';

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe('ChannelMenuItem', () => {

  it('create shapshot', async () => {
    const json = await import('./__jsons__/presenter.test.json');
    const params = {
      drive: json.default as Drive,
      onClick: jest.fn()
    };
    render(<DrivelMenuItem {...params} />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
