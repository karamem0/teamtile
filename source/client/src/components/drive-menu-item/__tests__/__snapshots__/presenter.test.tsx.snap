// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelMenuItem create shapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto auto;
  grid-gap: 0.25rem;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  cursor: pointer;
}

.emotion-1 {
  width: 1rem;
  height: 1rem;
  line-height: 1rem;
}

.emotion-2 {
  line-height: 0.75rem;
}

<body>
  <div>
    <span
      class="ui-text emotion-0"
      data-uses-unhanded-props="true"
      role="button"
    >
      <span
        class="emotion-1"
      >
        <span
          aria-hidden="true"
          class="root_dd790ee3"
        >
          <svg
            class="svg_dd790ee3"
            viewBox="0 0 2048 2048"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2048 1152q0 106-40 199t-110 162-163 110-199 41q-65 0-130-17-9 85-47 158t-99 128-137 84-163 31q-93 0-174-35t-142-96-96-142-36-175q0-16 1-32t4-32H85q-35 0-60-25t-25-60V597q0-35 25-60t60-25h302q12-109 62-202t127-163T751 39 960 0q119 0 224 45t183 124 123 183 46 224q0 16-1 32t-3 32q106 0 200 40t164 109 111 162 41 201zM960 128q-83 0-158 29t-135 80-98 122-52 153h422q35 0 60 25t25 60v422l18-3q18-64 52-121t80-104 104-81 122-52q8-43 8-82 0-93-35-174t-96-142-142-96-175-36zM522 1357q46 0 90-9t80-32 56-58 22-90q0-54-22-90t-57-60-73-39-73-29-56-29-23-38q0-17 12-27t28-17 35-8 30-2q51 0 90 13t81 39V729q-24-7-44-12t-41-8-41-5-48-2q-44 0-90 10t-83 32-61 58-24 88q0 51 22 85t57 58 73 40 73 31 56 31 23 39q0 19-10 30t-27 17-33 7-32 2q-60 0-106-20t-95-53v160q102 40 211 40zm438 563q66 0 124-25t101-68 69-102 26-125q0-57-19-109t-53-93-81-71-103-41v165q0 35-25 60t-60 25H646q-6 32-6 64 0 66 25 124t68 102 102 69 125 25zm576-384q79 0 149-30t122-82 83-122 30-150q0-79-30-149t-82-122-123-83-149-30q-79 0-148 30t-122 83-83 122-31 149v22q0 11 2 22 47 23 87 55t71 73 54 88 33 98q67 26 137 26z"
            />
          </svg>
        </span>
      </span>
      <span
        class="ui-text emotion-2"
        data-uses-unhanded-props="true"
      />
    </span>
  </div>
</body>
`;
